File: internal/content/testdata/bench/go/nc-diameter-tree.go
func diameter_of_binary_tree(root *TreeNode) int {
	best := 0
	var height func(node *TreeNode) int
	height = func(node *TreeNode) int {
		if node == nil {
			return 0
		}
		l := height(node.Left)
		r := height(node.Right)
		if l+r > best {
			best = l + r
		}
		if l > r {
			return 1 + l
		}
		return 1 + r
	}
	height(root)
	return best
}

File: internal/content/testdata/bench/java/nc-diameter-tree.java
class Solution {
    private long best = 0;
    public long diameter_of_binary_tree(TreeNode root) {
        best = 0;
        height(root);
        return best;
    }
    private long height(TreeNode node) {
        if (node == null) return 0;
        long l = height(node.left);
        long r = height(node.right);
        if (l + r > best) best = l + r;
        return 1 + Math.max(l, r);
    }
}
